### PR TITLE
"The Gang Does a Refactor"

### PR DESCRIPTION
**Pull request by j00bar**
Refactoring Turnpike to be based on more pluggable components in the policy service.

1. Restructures repository to be Python centric.
2. Adapts the Flask app to be more modular, including a plugin system for processing requests.
3. Implements a couple of extra plugins for funsies.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -110,6 +110,23 @@ your Python expression could be:
 The evaluation would use set-logic to look for overlaps. If there were any overlaps, the predicate would evaluate to
 `True`. If not, `False`.
 
+Customizing and Extending Turnpike
+----------------------------------
+
+Turnpike's policy service is based on passing a request through a series of plugins that
+evaluate the request for conformity with the deployment policies. These plugins allows
```
Comment (coderbydesign) [inline docs/index.md:117]:
```suggestion
evaluate the request for conformity with the deployment policies. These plugins allow
```

File: turnpike/config.py [hunk 2]
```diff
@@ -17,3 +18,12 @@
 )
 PERMANENT_SESSION_LIFETIME = 60 * 60 * 4
 SESSION_COOKIE_SECURE = True
+
+PLUGIN_CHAIN = ["turnpike.plugins.auth.AuthPlugin", "turnpike.plugins.rh_identity.RHIdentityPlugin"]
```
Comment (coderbydesign) [inline turnpike/config.py:22]: Would it make sense to have these plugin definitions configurable?

**Review by coderbydesign — COMMENTED** [review]
🚀 This is great stuff! Tested locally and everything is working as expected. Just had two minor comments.

File: turnpike/config.py (hunk 2 above)
Comment (j00bar) [reply turnpike/config.py]: Created the ability for deployers to override config in a change today.

**Review by coderbydesign — APPROVED** [review]

**Pull request merged**